sdk/src/Core/Amazon.Runtime/Credentials/Internal/IdentityResolvers/DefaultAWSTokenIdentityResolver.cs

```diff
@@ -0,0 +1,60 @@
+﻿/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+using Smithy.Identity.Abstractions;
+using System;
+
+namespace Amazon.Runtime.Credentials.Internal.IdentityResolvers
+{
+    /// <summary>
+    /// A resolver that provides an AWSToken identity. 
+    /// </summary>
+    public class DefaultAWSTokenIdentityResolver : IIdentityResolver
+    {
+        /// <summary>
+        /// Gets or sets the AWSTokenProvider used to resolve the AWSToken.
+        /// By default, this is initialized with an AWSTokenProviderChain containing a ProfileTokenProvider.
+        /// </summary>
+        public IAWSTokenProvider AWSTokenProvider { get; set; }
+
+        public DefaultAWSTokenIdentityResolver()
+        {
+            AWSTokenProvider = new AWSTokenProviderChain(new ProfileTokenProvider());
+        }
+
+        public BaseIdentity ResolveIdentity()
+        {
+            AWSToken token;
+#if BCL
```

Comment:
This kinda pushes us to create an `async` version of `ResolveIdentity`, we can always add it later if we don't want to diverge allot from the SRA.

Reply:
We should not do this (look how long it took to fix the blocking call we had in the DynamoDB HLL). Honestly I'd prefer if we did all SSO related changes in its PR (since we may have to adapt the existing components we have in the SDK).